plugins/session_notifier.rb

```diff
@@ -315,6 +333,30 @@ def send_text_to_gotify(session)
         end
       end
 
+      def send_text_to_serverjang(session)
+        # https://sct.ftqq.com/sendkey
+        platform = "#{session.platform}"
+        arch = "#{session.arch}"
+        tunnel = "#{session.tunnel_to_s}"
+        uri_parser = URI.parse(serverjang_webhook)
+        params = {}
+        params["title"] = "You have new #{session.type} session"
+        params["desp"] = 'OS:%s, tunnel:%s, Arch:%s' %[platform,tunnel,arch]
+        http = Net::HTTP.new(uri_parser.host, uri_parser.port)
+        http.use_ssl = true
+        res = Net::HTTP::post_form(uri_parser,params)
+        body = JSON.parse(res.body)
+        if res.nil? || res.body.blank?
+          print_error("No response recieved from the ServerJang server!")
+          return nil
```

Comment:
All of this honestly should be moved under the `begin` statement below as this is what the `JSON::ParserError` is designed to catch: exceptions from `JSON.parse` being called with incorrect JSON data.

Additionally the order of these checks is wrong as you are using `res.body` and then only after are you checking if `res` is `nil` or `res.body` is empty. So the order of these checks needs to be redone.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/16192/commits/5a8cf3ea95aea77f220a8548c89535868fbc0c34